cpp/src/arrow/gpu/cuda_memory.cc

```diff
@@ -501,5 +502,27 @@ Result<std::shared_ptr<MemoryManager>> DefaultMemoryMapper(ArrowDeviceType devic
   }
 }
```

Comment:
Do we still need this `DefaultMemoryMapper` function?

Reply:
I can remove `DefaultMemoryMapper`, yes (to be clear it's not actually used right now, only exposed. But given that a user would also not need it anymore to pass to the `Import..` functions, I assume it can indeed be removed)

Reply:
Strictly speaking this is publicly exposed in a released version. Do we first need to deprecate it, or given that this is brand new it's OK to still remove directly?
(it was added in 14.0)

Reply:
Ok, let's deprecate it then?